src/stdlib_strings.f90

```diff
@@ -649,4 +669,122 @@ pure function replace_all_char_char_char(string, pattern, replacement) result(re
+        integer, intent(in) :: output_length
+        character(len=max(len(string), output_length)) :: res
+
+        res = padl(string, output_length, " ")
+
+    end function padl_char_default
+
+    !> Left pad the input string with the 'pad_with' character
+    !>
+    !> Returns a new string
+    pure function padl_char_pad_with(string, output_length, pad_with) result(res)
+        character(len=*), intent(in) :: string
+        integer, intent(in) :: output_length
+        character(len=1), intent(in) :: pad_with
+        character(len=max(len(string), output_length)) :: res
+        integer :: string_length
+
+        string_length = len(string)
+
+        if (string_length < output_length) then
+            res = repeat(pad_with, output_length - string_length)
+            res(output_length - string_length + 1 : output_length) = string
+        else
+            res = string
+        end if
+
+    end function padl_char_pad_with
+
+    !> Right pad the input string with " " (1 whitespace)
+    !>
+    !> Returns a new string
+    pure function padr_string_default(string, output_length) result(res)
+        type(string_type), intent(in) :: string
+        integer, intent(in) :: output_length
+        character(len=max(len(string), output_length)) :: char_output
+        type(string_type) :: res
+
+        char_output = char(string)
+        res = string_type(char_output)
+
+    end function padr_string_default
+
+    !> Right pad the input string with the 'pad_with' character
+    !>
+    !> Returns a new string
+    pure function padr_string_pad_with(string, output_length, pad_with) result(res)
+        type(string_type), intent(in) :: string
+        integer, intent(in) :: output_length
+        character(len=1), intent(in) :: pad_with
+        type(string_type) :: res
+
+        res = string_type(padr(char(string), output_length, pad_with))
+
+    end function padr_string_pad_with
+
+    !> Right pad the input string with " " (1 whitespace)
+    !>
+    !> Returns a new string
+    pure function padr_char_default(string, output_length) result(res)
+        character(len=*), intent(in) :: string
+        integer, intent(in) :: output_length
+        character(len=max(len(string), output_length)) :: res
+
+        res = string
+
+    end function padr_char_default
+
+    !> Right pad the input string with the 'pad_with' character
+    !>
+    !> Returns a new string
+    pure function padr_char_pad_with(string, output_length, pad_with) result(res)
+        character(len=*), intent(in) :: string
+        integer, intent(in) :: output_length
+        character(len=1), intent(in) :: pad_with
+        character(len=max(len(string), output_length)) :: res
+        integer :: string_length
+
+        string_length = len(string)
+
+        res = string
+        if (string_length < output_length) then
+            res(string_length + 1 : output_length) = repeat(pad_with, &
+            & output_length - string_length)
```

Comment:
I find the line break in the middle of function arguments list somewhat awkward, what do you think about breaking it at the assignment operator, i.e. LHS on one line and RHS on the other?

```suggestion
            res(string_length + 1 : output_length) = &
              repeat(pad_with, output_length - string_length)
```

I noticed that you use the continuation symbol `&` both at the end of the continuing line and at the beginning of the continued line. It's not necessary, you can have it only on the continuing line. But maybe you use both as a visual aid, which I think is fine.